src/Storage/Storage.Management/StorageAccount/RestoreAzStorageBlobRange.cs

```diff
@@ -119,49 +121,80 @@ public override void ExecuteCmdlet()
+                    }
 
-                    Task<AzureOperationResponse<BlobRestoreStatus>> waitTask = ((StorageManagementClient)this.StorageClient).GetPostOrDeleteOperationResultAsync(response, null, new System.Threading.CancellationToken());
                     try
                     {
-                        waitTask.Wait();
+                        var result = restoreLro.WaitForCompletion().Value;
+                        WriteObject(new PSBlobRestoreStatus(result));
                     }
-                    catch (System.AggregateException ex) when (ex.InnerException is CloudException)
+                    catch (Exception ex)
                     {
-                        throw new InvalidJobStateException(string.Format("Blob ranges restore failed with information: '{0}'.", ((CloudException)ex.InnerException).Response.Content));
+                        throw new InvalidJobStateException(string.Format("Blob ranges restore failed with information: '{0}'.", ex.ToString()));
                     }
-
-                    AzureOperationResponse<BlobRestoreStatus> result = waitTask.Result;
-
-                    WriteObject(new PSBlobRestoreStatus(result.Body));
-
                 }
                 else
                 {
-                    BlobRestoreStatus status = this.StorageClient.StorageAccounts.BeginRestoreBlobRanges(
-                     this.ResourceGroupName,
-                     this.StorageAccountName,
-                     this.TimeToRestore,
-                     PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange));
-
-                    WriteObject(new PSBlobRestoreStatus(status));
-                    if (status != null && status.Status == BlobRestoreProgressStatus.Failed)
+                    Track2.StorageAccountResource account = this.StorageClientTrack2.GetStorageAccount(this.ResourceGroupName, this.StorageAccountName);
+                    var restoreLro = account.RestoreBlobRanges(
+                        WaitUntil.Completed,
+                        new Track2Models.BlobRestoreContent(
+                            this.TimeToRestore.ToUniversalTime(),
+                            ParseBlobRestoreRanges(this.BlobRestoreRange)));
+                    if (restoreLro.Value != null && restoreLro.Value.Status != null && restoreLro.Value.Status == Track2Models.BlobRestoreProgressStatus.Failed)
                     {
                         throw new InvalidJobStateException("Blob ranges restore failed.");
                     }
+                    WriteObject(new PSBlobRestoreStatus(restoreLro.Value));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parse from PSBlobRestoreRange list to BlobRestoreRange list
+        /// </summary>
+        public static IList<Track2Models.BlobRestoreRange> ParseBlobRestoreRanges(PSBlobRestoreRange[] ranges)
```

Comment:
This should be a function of PSBlobRestoreRange.

Reply:
Done. Didn't notice there's already a ParseBlobRestoreRanges in PSBlobRestoreRange. Therefore simply removing this duplicated function here.